File: MorskoGram.Web.ViewModels/Users/UserViewModel.cs
namespace MorskoGram.Web.ViewModels.Users
{
    using System.Collections.Generic;
    using MorskoGram.Data.Models;
    using MorskoGram.Services.Mapping;
    using MorskoGram.Web.ViewModels.Follows;

    public class UserViewModel : IMapFrom<ApplicationUser>
    {
        public string Id { get; set; }

        public string Email { get; set; }

        public ICollection<FollowViewModel> Followers { get; set; }

        public ICollection<FollowViewModel> Followings { get; set; }
    }
}

File: MorskoGram.Web.API/ClientApp/src/models/users/UserViewModel.js
import {Mergeable} from '../Mergeable';

export class UserViewModel extends Mergeable {
    constructor(obj) {
        super(obj);

        if (!obj) {
            this.id = '';
            this.email = '';
            /**
             * @type {FollowViewModel[]}
             */
            this.followers = [];
            /**
             * @type {FollowViewModel[]}
             */
            this.followings = [];
        }
    }
}
